stm32: nucleo_g474re: reflash board successively with pyocd

FRASTM opened the pull request:
It happens that the nucleo_g474re board  cannot be flashed a second time and reboot with pyocd.
(copy binary to the target board is still functional)
This patch changes the board_runner_args(pyocd "--target=stm32g474rbtx") in the board.cmake 
Use west flash to program the target board /zephyr/boards/arm/nucleo_g474re

Signed-off-by: Francois Ramu francois.ramu@st.com

Fixes #23351

alexanderwachter commented on boards/arm/nucleo_g474re/board.cmake:
```diff
@@ -1,4 +1,4 @@
 # SPDX-License-Identifier: Apache-2.0
-board_runner_args(pyocd "--target=stm32g474retx")
+board_runner_args(pyocd "--target=stm32g474rbtx")
```
This change works for me! Thanks

erwango commented on boards/arm/nucleo_g474re/board.cmake:
```diff
@@ -1,4 +1,4 @@
 # SPDX-License-Identifier: Apache-2.0
-board_runner_args(pyocd "--target=stm32g474retx")
+board_runner_args(pyocd "--target=stm32g474rbtx")
```
This would deserve a comment so we know this is made on purpose, to solve a flashing issue.
Also, are we sure this has no impact otherwise (which should be advertised as limitations )?

FRASTM replied:
adding a comment

FRASTM replied:
adding a comment

erwango commented on soc/arm/st_stm32/stm32g4/Kconfig.defconfig.stm32g474re:
```diff
@@ -6,11 +6,9 @@
 
 config NUM_IRQS
-	int
 	default 102
```
Let's split this is a dedicated commit (same PR could be used).

FRASTM replied:
done into 13c405dda1e2f443a592a5464120fdf04cddbf84 and  3611b5da65c826dd2d62ac3d2517f316a0670040

FRASTM commented:
separate in 2 commits

FRASTM committed:
board: arm: nucleo_g474re reflashing board

Reflash the nucleo_g474re successively

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM committed:
soc: arm: stm32g474 config device

This patch aligns the Kconfig.defconfig.stm32g474re to the
Kconfig.defconfig.stm32g431rb

Signed-off-by: Francois Ramu <francois.ramu@st.com>

FRASTM commented:
adding a comment in boards/arm/nucleo_g474re/board.cmake
